easybuild/easyblocks/b/boost.py

```diff
@@ -175,6 +176,10 @@ def build_step(self):
         bjamoptions = " --prefix=%s" % self.objdir
 
         cxxflags = os.getenv('CXXFLAGS')
+        if self.cfg['use_glibcxx11_abi'] is False:
+            cxxflags += ' -D_GLIBCXX_USE_CXX11_ABI=0'
+        elif self.cfg['use_glibcxx11_abi']:
+            cxxflags += ' -D_GLIBCXX_USE_CXX11_ABI=1'
```

Comment:
@smoors This changes the default to always include `-D_GLIBCXX_USE_CXX11_ABI=1`, is that a good idea?

Reply:
That's not the idea. If the flag is not set, the default setting is used, whatever that is (in that case `self.cfg['use_glibcxx11_abi'] is None`).
If it is set to True/False, it is used/not used regardless of the default setting.

Reply:
Or maybe I misunderstood how this works in easybuild?

Reply:
No, you're right, but the logic is a bit confusing, and the duplicate code doesn't help.

How about this as alternative:

```python
cxxflags += ' -D_GLIBCXX_USE_CXX11_ABI='
if self.cfg['use_glibcxx11_abi']:
    cxxflags += '1'
else:
    cxxflags += '0'
```

Reply:
@boegel Your alternative is clearer but does not allow for the default GCC setting to be used. Maybe that's ok because `-D_GLIBCXX_USE_CXX11_ABI=1` has been the default since GCC 5 ?
I suggest another even simpler alternative (with comment added):
```
# If 'use_glibcxx11_abi' is not set or is True, use the default which is '-D_GLIBCXX_USE_CXX11_ABI=1' since GCC 5
if self.cfg['use_glibcxx11_abi'] is False:
    cxxflags += ' -D_GLIBCXX_USE_CXX11_ABI=0'
```

Reply:
You're right w.r.t. the default. Your alternative makes sense, but doesn't give the option to explicitly enable it, which may be useful when working with older GCC versions.

So:

```python
cxxflags += ' -D_GLIBCXX_USE_CXX11_ABI='
if self.cfg['use_glibcxx11_abi']:
    cxxflags += '1'
elif self.cfg['use_glibcxx11_abi'] is False:
    # only disable if use_glibcxx11_abi was explicitly set to False
    # None value is the default, which corresponds to default setting (=1 since GCC 5.x)
    cxxflags += '0'

Reply:
I agree. Note that the logic is now the same as my original code ;)

Reply:
If it is None then wouldn't it ignore both cases and end up with ...ABI= without a 0 or 1 at the end?

(Python is still not my friend)

Reply:
Ah, yes :(

Reply:
Good catch @akesandgren, so maybe wrap the whole thing under `if self.cfg['use_glibcxx11_abi'] is not None:`?

Reply:
too late, I already did it ;)